fe1-web/src/features/evoting/components/EventElection.tsx

```diff
@@ -37,55 +32,21 @@ const styles = StyleSheet.create({
   } as TextStyle,
 });
 
-const EventElection = (props: IPropTypes) => {
+const EventElection: FunctionComponent<IPropTypes> = (props) => {
   const { election } = props;
   const { isOrganizer } = props;
   const toast = useToast();
-  const questions = election.questions.map((q) => ({ title: q.question, data: q.ballot_options }));
-  const [selectedBallots, setSelectedBallots] = useState(new Array(questions.length).fill([]));
-  const [hasVoted, setHasVoted] = useState(0);
-  const untilStart = (election.start.valueOf() - Timestamp.EpochNow().valueOf()) * 1000;
-  const untilEnd = (election.end.valueOf() - Timestamp.EpochNow().valueOf()) * 1000;
-  // This makes sure the election status is always updated
-  const electionFromStore = useSelector(
-    (state) =>
-      // @ts-ignore
-      state.events.byLaoId[election.lao].byId[election.id],
+  const questions = useMemo(
+    () => election.questions.map((q) => ({ title: q.question, data: q.ballot_options })),
+    [election.questions],
   );
-
-  if (!electionFromStore) {
-    throw new Error('Error in Election display: Election doesnt exist in store');
-  }
-
-  const updateSelectedBallots = (values: number[], idx: number) => {
-    setSelectedBallots((prev) => prev.map((item, id) => (idx === id ? values : item)));
-  };
-  const concatenateIndexes = (indexes: number[]) => {
-    let concatenated = '';
-    indexes.forEach((index) => {
-      concatenated += index.toString();
-    });
-    return concatenated;
-  };
-
-  // Prepares the votes with the hash and the vote indexes to match the protocol
-  // id: SHA256('Vote'||election_id||question_id||(vote_index(es)|write_in))
-  const refactorVotes = (selected: number[][]) => {
-    const votes: Vote[] = selected.map((item, idx) => ({
-      id: Hash.fromStringArray(
-        EventTags.VOTE,
-        election.id.toString(),
-        election.questions[idx].id,
-        concatenateIndexes(item),
-      ).valueOf(),
-      question: election.questions[idx].id,
-      vote: item,
-    }));
-    return votes;
-  };
+  const [selectedBallots, setSelectedBallots] = useState<{ [questionIndex: number]: Set<number> }>(
```

Comment:
I would strongly suggest creating a type for the state. It would help with documenting its expected values.